src/k8s-engine/helm.js

```
@@ -0,0 +1,113 @@
+'use strict';
+
+const { spawn } = require('child_process');
+const os = require('os');
+
+/*
+ * List returns the current Helm releases in a namespace. If no namespace is
+ * provided the current default is used. It is recommended that you
+ */
+async function list(namespace) {
+  return new Promise((resolve, reject) => {
+    let dta, err
+    let args = ['ls', '--kube-context', 'rancher-desktop', '-o', 'json']
+    if (namespace != undefined) {
+      args.push('-n')
+      args.push(namespace)
+    }
+    const bat = spawn('./resources/' + os.platform() + '/bin/helm', args)
+
+    bat.stdout.on('data', (data) => {
+      dta = data.toString()
+    })
+
+    bat.stderr.on('data', (data) => {
+      err = data.toString()
+    })
+
+    bat.on('exit', (code) => {
+      console.log('code ' + code)
+      if (code === 0) {
+        resolve(dta)
+      } else {
+        reject('Failed to list resource: ' + err)
+      }
+    })
+  })
+}
+
+async function status(name, namespace) {
+  return new Promise((resolve, reject) => {
+    if (name === undefined) {
+      reject("name required to get status")
+    }
+    
+    let dta, err
+    let args = ['status', name, '--kube-context', 'rancher-desktop', '-o', 'json']
+    if (namespace != undefined) {
+      args.push('-n')
+      args.push(namespace)
+    }
+
+    const bat = spawn('./resources/' + os.platform() + '/bin/helm', args)
+
+    bat.stdout.on('data', (data) => {
+      dta = data.toString()
+    })
+
+    bat.stderr.on('data', (data) => {
+      err = data.toString()
+    })
+
+    bat.on('exit', (code) => {
+      console.log('code ' + code)
+      if (code === 0) {
+        resolve(dta)
+      } else {
+        reject('Failed to list resource: ' + err)
+      }
+    })
+  });
+}
+
+async function install(name, chart, namespace) {
+  return new Promise((resolve, reject) => {
+    if (name === undefined) {
+      reject("name required to install")
```

Yeah, TypeScript would be nice here… (Not clear why this needs to be inside the promise, but ¯\\\_(ツ)\_/¯).

---

It technically does not need to be in the promise. If outside of it the code would need to throw an exception or use `Promise.reject`.